runtime/src/main/java/org/corfudb/runtime/LiteRoutingQueueListener.java

```diff
@@ -72,118 +74,129 @@ public LiteRoutingQueueListener(CorfuStore corfuStore, String sourceSiteId, Stri
+                .findFirst().get();
+
         List<Table.CorfuQueueRecord> allEntries = new ArrayList<>(entries.size());
-        Queue.ReplicationType currentType = Queue.ReplicationType.LAST_FULL_SYNC_ENTRY;
+        List<Queue.RoutingTableEntryMsg> allRQMsgs = new ArrayList<>();
+
         for (CorfuStreamEntry entry : entries) {
-            if (!entry.getOperation().equals(CorfuStreamEntry.OperationType.UPDATE)) {
+            // The Source always 'adds' replicated data to the queue.  So ignore all operations where type != UPDATE.
+            // Non-update entries are written by LR (CLEAR) and this listener itself (DELETE after successful
+            // processing).
+            if (entry.getOperation() != CorfuStreamEntry.OperationType.UPDATE) {
                 continue;
             }
             Queue.CorfuGuidMsg key = (Queue.CorfuGuidMsg) entry.getKey();
             Queue.RoutingTableEntryMsg msg = (Queue.RoutingTableEntryMsg) entry.getPayload();
+            allRQMsgs.add(msg);
+
             Queue.CorfuQueueMetadataMsg metadataMsg = (Queue.CorfuQueueMetadataMsg) entry.getMetadata();
             allEntries.add(new Table.CorfuQueueRecord(key, metadataMsg, msg));
-            if (currentType.equals(Queue.ReplicationType.LAST_FULL_SYNC_ENTRY)) {
-                currentType = msg.getReplicationType();
-            } else if (!currentType.equals(msg.getReplicationType())) {
-                throw new LogReplicationClientException("Not expecting mixed event types "+ currentType +" vs "
-                    +msg.getReplicationType());
-            }
+            currentReplicationType = msg.getReplicationType();
         }
         if (allEntries.isEmpty()) {
             return;
         }
-        boolean entriesProcessed;
+        int numEntriesProcessed;
         long now = System.currentTimeMillis();
-        long whenQrecordWasCreated = CorfuGuid.getTimestampFromGuid(allEntries.get(0).getRecordId().getInstanceId(), now);
-        if (currentType.equals(Queue.ReplicationType.LOG_ENTRY_SYNC)) {
-            log.info("LRQRecvListener delivering {} LOG_ENTRY updates took {}ms end to end",
-                    allEntries.size(), now - whenQrecordWasCreated);
-            entriesProcessed = processUpdatesInLogEntrySync(allEntries.stream().sorted().map(q ->
-                    (Queue.RoutingTableEntryMsg)q.getEntry()).collect(Collectors.toList()));
-        } else {
-            log.info("LRQRecvListener delivering {} FULLSYNC updates took {}ms end to end",
-                    allEntries.size(), now - whenQrecordWasCreated);
-            entriesProcessed = processUpdatesInSnapshotSync(allEntries.stream().sorted().map(q ->
-                    (Queue.RoutingTableEntryMsg)q.getEntry()).collect(Collectors.toList()));
-        }
+        long whenQRecordWasCreated = CorfuGuid.getTimestampFromGuid(allEntries.get(0).getRecordId().getInstanceId(),
```

Comment:
sorry for confusion but can we just remove these `whenQRecordWasCreated` - with the latest queue changes the timestamp wont be part of the queue anyway.